tools/summaries/palm.py

```diff
@@ -12,49 +12,112 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import json
+from typing import List
+
 import logging
 import os
-import re
-
-import dc
 import requests
 
 _API_KEY = os.getenv("PALM_API_KEY")
-_API_URL = "https://generativelanguage.googleapis.com/v1beta2/models/text-bison-001:generateText"
+_API_URL = "https://generativelanguage.googleapis.com/v1beta3/models/text-bison-001:generateText"
 _TEMPERATURE = 0.2
-# Sometimes the summaries may not be factual. Retry them in that case.
-# We use a heuristic that the summary must contain at least 2 numbers to be factual.
-_RETRIES = 2
+
+_SERIES_PROMPT = """
+Generate a summary in 2 sentences using only the information from the following tables.
+Only list important highlights per table.
+The summary should only be based on the information presented in these tables.
+Do not include facts from other sources.
+Do not use superlatives.
+Do not use the phrase 'According to the data'.
+Do not include opinions.
+Please include references if information is included from other sources.
+Please write in a professional and business-neutral tone.
+
+{place_type}: {place_name}
+
+{ranking_key}:
+{ranking_data}
+
+Table:
+{data_table}
+
+Summary:
+"""
+
+_RESUMMARIZE_PROMPT = """
+Summarize these facts into 1 paragraph.
+Start by introducing the place.
+The summary should only be based on the information presented in these facts.
+Please write in a professional and business-neutral tone.
+
+Facts:
+{facts}
+
+Summary:
+"""
 
 assert _API_KEY, "$PALM_API_KEY must be specified."
 
+# Ranking key -> data_table key
+_TABLE_KEYS = {
+  "Largest Population": "Count_Person",
+  "Highest Median Income": "Median_Income_Person",
+  "Highest Median Age": "Median_Age_Person",
+}
+
 
-def get_summary(name: str, csv: str):
+def strip_superlatives(label: str) -> str:
+  label = label.replace("Highest", "")
+  label = label.replace("Largest", "")
```

Comment:
nit: Can you add a comment on what is being stripped out? Like "Highest Income" to "Income"?

Reply:
done